Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp

```diff
@@ -947,4 +1011,179 @@ namespace AZ::Render
             }
         }
     }
+
+    // Find the trajectory path for a given actor instance
+    AtomActorDebugDraw::TrajectoryTracePath* AtomActorDebugDraw::FindTrajectoryPath(const EMotionFX::ActorInstance* actorInstance)
+    {
+        for (TrajectoryTracePath* trajectoryPath : m_trajectoryTracePaths)
+        {
+            if (trajectoryPath->m_actorInstance == actorInstance)
+            {
+                return trajectoryPath;
+            }
+        }
+
+        // we haven't created a path for the given actor instance yet, do so
+        TrajectoryTracePath* tracePath = new TrajectoryTracePath();
+
+        tracePath->m_actorInstance = actorInstance;
+        tracePath->m_traceParticles.reserve(512);
+
+        m_trajectoryTracePaths.emplace_back(tracePath);
+        return tracePath;
+    }
+
+    void AtomActorDebugDraw::RenderTrajectoryPath(AzFramework::DebugDisplayRequests* debugDisplay,
+        const EMotionFX::ActorInstance* actorInstance,
+        const AZ::Color& headColor,
+        const AZ::Color& pathColor)
+    {
+        TrajectoryTracePath* trajectoryPath = FindTrajectoryPath(actorInstance);
+        if (!trajectoryPath)
+        {
+            return;
+        }
+
+        EMotionFX::Actor* actor = actorInstance->GetActor();
+        EMotionFX::Node* extractionNode = actor->GetMotionExtractionNode();
+        if (!extractionNode)
+        {
+            return;
+        }
+
+        // Fast access to the trajectory trace particles
+        const AZStd::vector<TrajectoryPathParticle>& traceParticles = trajectoryPath->m_traceParticles;
+        const size_t numTraceParticles = traceParticles.size();
+        if (traceParticles.empty())
+        {
+            return;
+        }
+
+        const float trailWidthHalf = 0.25f;
+        const float trailLength = 2.0f;
+        const float arrowWidthHalf = 0.75f;
+        const float arrowLength = 1.5f;
```

Comment:
suggestion: Could also be candidates for AZ_CVARs to let designers tweak the look more easily in future, can be quite handy and pretty quick to do

Reply:
We can probably added it to the preference window of emfx, but honestly I think that is a bit overkill? I don't think people care much about what size the arrow head length is tbh.
AZ_CVARs should reserved for more valuable variables imo.

Reply:
Sure thing, it's just if a designer does want to start customizing things these can be helpful (that we hide from users, so it's not a setting exactly, but useful to tweak if people want to say change the size/color etc...)